core/src/test/java/io/grpc/internal/TestUtils.java

```diff
@@ -36,7 +41,41 @@
 /**
  * Common utility methods for tests.
  */
-final class TestUtils {
+public final class TestUtils {
```

Comment:
This is wrong. This in `core/src/`**test**`/java/io/grpc/internal/TestUtils.java`. It is a package-private utility class for `grpc-core`'s unit test only. Not a test utility.

Reply:
The javadoc is specifying it's a test utility. I'm extending it to be public to the entire project.

Reply:
It's a test utility for `grpc-core`'s unit tests, not for other modules. Look at its file path.

Reply:
We did the same for `FakeClock`. We can not put it in `main` sourceSet, but we want all modules to use it. `grpc-core` is the most reasonable module for all modules to use.

Reply:
I am surprised `FakeClock` is from `grpc-core`'s test scope.

I looked at its history. I suspect initially it was created for testing classes in `grpc-core`'s `io.grpc.internal` package (#1282). It was package-private at that time. Then it was used for testing classes in `grpc-core`'s `io.grpc` package (#1602) and its was made to be public. This usage was still fair as no cross-module happened. After that, people started to use (abused easily as it is public) it with cross-module reference and put the hack (`project(':grpc-core').sourceSets.test.output`) in build files.

I filed #6847 to address this issue (and we may discuss it in next API meeting).

Anyway, using classes from another module's test scope is bad, it's access is hacked. You should not do this. Also, as I mentioned in the next comment, in your case the usage is completely unnecessary.